EIPS/eip-7066.md

```diff
@@ -0,0 +1,157 @@
+get purged.
+
+Token MAY be transfered and `locked`, and OPTIONAL retain the value of `approval` using `transferAndLock` function. This is RECOMMENDED for usecases where Token transfer and subsequent revocation is REQUIRED.
+
+### Interface
+
+```
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >=0.7.0 <0.9.0;
+
+/// @title Lockable Extension for ERC721
+/// @dev Interface for the Lockable extension
+/// @author StreamNFT 
+
+interface IERC7066{
+
+    /**
+     * @dev Emitted when tokenId is locked
+     */
+    event Lock (uint256 indexed tokenId, address _locker);
+
+    /**
+     * @dev Emitted when tokenId is unlocked
+     */
+    event Unlock (uint256 indexed tokenId);
+
+    /**
+     * @dev Lock the tokenId if msg.sender is owner or approved and set locker to msg.sender
+     */
+    function lock(uint256 tokenId) external;
+
+    /**
+     * @dev Lock the tokenId if msg.sender is owner and set locker to _locker
+     */
+    function lock(uint256 tokenId, address _locker) external;
+
+    /**
+     * @dev Unlocks the tokenId if msg.sender is locker
+     */
+    function unlock(uint256 tokenId) external;
+
+    /**
+     * @dev Tranfer and lock the token if the msg.sender is owner or approved. 
+     *      Lock the token and set locker to caller
+     *      Optionally approve caller if bool setApprove flag is true
+     */
+    function transferAndLock(uint256 tokenId, address from, address to, bool setApprove) external;
+
+    /**
+     * @dev Returns the wallet, that is stated as unlocking wallet for the tokenId.
+     *      If address(0) returned, that means token is not locked. Any other result means token is locked.
+     */
+    function lockerOf(uint256 tokenId) external view returns (address);
+}
+```
+
+## Rationale
+
+This approach presents a minimalistic solution that focuses on locking items and specifying who has the authority to unlock them. It offers flexibility and extensibility, accommodating various potential use cases mentioned in the Motivation section.
+
+Moreover, when there is a requirement to grant temporary or redeemable rights for a NFT, such as rentals or purchases with installments, this EIP involves the actual transfer of the token to the temporary user's wallet, rather than simply assigning a role. This design choice ensures compatibility with existing NFT ecosystem tools and dApps,  without necessitating additional interfaces or logic implementation.
+
+This functionality already exists on Solana, enabling ease for NFT liquidity and use cases. This EIP shall introduce same functionality to EVM ecosystem. The naming and reference implementation of functions and storage entities resemble the Approval flow outlined in [ERC-721](./eip-721.md), ensuring an intuitive user experience. 
+
+Existing Upgradedable [ERC-721](./eip-721.md) can upgrade to this standard, enabling locking capability inherently and unlock underlying liquidity features.
+
```

Comment:
Might be nice to explain why `lockerOf` returns the zero address instead of reverting when the token is not locked.